go/libraries/doltcore/schema/encoding/schema_marshaling.go

```diff
@@ -278,16 +323,54 @@ func MarshalSchemaAsNomsValue(ctx context.Context, vrw types.ValueReadWriter, sc
 	return types.EmptyStruct(vrw.Format()), errors.New("Table Schema could not be converted to types.Struct")
 }
 
+type schCacheData struct {
+	all   *schema.ColCollection
+	pk    *schema.ColCollection
+	nonPK *schema.ColCollection
+	sd    *schemaData
+}
+
+var unmarshalledSchemaCache = map[hash.Hash]schCacheData{}
+
 // UnmarshalSchemaNomsValue takes a types.Value instance and Unmarshalls it into a Schema.
 func UnmarshalSchemaNomsValue(ctx context.Context, nbf *types.NomsBinFormat, schemaVal types.Value) (schema.Schema, error) {
```

Comment:
If we changed this signature to take the Schema's `types.Ref`, cache hits would eliminate a chunk read